src/lib.rs

```diff
@@ -98,7 +98,7 @@ fn json_set(ctx: &Context, args: Vec<String>) -> RedisResult {
 
     let mut format = Format::JSON;
     let mut set_option = SetOptions::None;
-    let mut index = None;
+    let mut value_index = None;
 
     loop {
```

Comment:
Not related to your change, but `cargo clippy` rightly suggests to simplify this `loop` with `while let`:

```
    while let Some(s) = args.next() {
        match s.to_uppercase().as_str() {
```

Reply:
:+1: